File: 08-august/14-maximum-length-substring-with-two-occurrences/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var maximumLengthSubstring = function(s) {
    const counts = new Array(26).fill(0);
    let left = 0;
    let answer = 0;

    for (let right = 0; right < s.length; right++) {
        const idx = s.charCodeAt(right) - 97;
        counts[idx]++;

        while (counts[idx] > 2) {
            counts[s.charCodeAt(left) - 97]--;
            left++;
        }

        answer = Math.max(answer, right - left + 1);
    }

    return answer;
};

File: 08-august/14-maximum-length-substring-with-two-occurrences/solution.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int[] counts = new int[26];
        int left = 0;
        int answer = 0;

        for (int right = 0; right < s.length(); right++) {
            int idx = s.charAt(right) - 'a';
            counts[idx]++;

            while (counts[idx] > 2) {
                counts[s.charAt(left) - 'a']--;
                left++;
            }

            answer = Math.max(answer, right - left + 1);
        }

        return answer;
    }
}
